cores/esp32/HWCDC.cpp

```diff
@@ -238,26 +239,33 @@ void HWCDC::begin(unsigned long baud)
         }    
     }
     usb_serial_jtag_ll_disable_intr_mask(USB_SERIAL_JTAG_LL_INTR_MASK);
-    usb_serial_jtag_ll_clr_intsts_mask(USB_SERIAL_JTAG_LL_INTR_MASK);
     usb_serial_jtag_ll_ena_intr_mask(USB_SERIAL_JTAG_INTR_SERIAL_IN_EMPTY | USB_SERIAL_JTAG_INTR_SERIAL_OUT_RECV_PKT | USB_SERIAL_JTAG_INTR_BUS_RESET);
     if(!intr_handle && esp_intr_alloc(ETS_USB_SERIAL_JTAG_INTR_SOURCE, 0, hw_cdc_isr_handler, NULL, &intr_handle) != ESP_OK){
         isr_log_e("HW USB CDC failed to init interrupts");
         end();
         return;
     }
-    if (perimanSetBusDeinit(ESP32_BUS_TYPE_USB_DM, HWCDC::deinit) && perimanSetBusDeinit(ESP32_BUS_TYPE_USB_DP, HWCDC::deinit)) {
-        // Setting USB D+ D- pins
-        perimanSetPinBus(USB_DM_GPIO_NUM, ESP32_BUS_TYPE_USB_DM, (void *) this, -1, -1);
-        perimanSetPinBus(USB_DP_GPIO_NUM, ESP32_BUS_TYPE_USB_DP, (void *) this, -1, -1);
-    } else {
-        log_e("Serial JTAG Pins can't be set into Peripheral Manager.");
-    }
+    // Setting USB D+ D- pins
+    uint8_t pin = ESP32_BUS_TYPE_USB_DM;
+    if(perimanGetPinBusType(pin) != ESP32_BUS_TYPE_INIT)
+        if(!perimanClearPinBus(pin)) goto err;
+    if(!perimanSetPinBus(pin, ESP32_BUS_TYPE_USB_DM, (void *) this, -1, -1)) goto err;
+    pin = ESP32_BUS_TYPE_USB_DP;
+    if(perimanGetPinBusType(pin) != ESP32_BUS_TYPE_INIT)
+        if(!perimanClearPinBus(pin)) goto err;
+    if(!perimanSetPinBus(pin, ESP32_BUS_TYPE_USB_DP, (void *) this, -1, -1)) goto err;
+    return;
+    
+    err:
+    log_e("Serial JTAG Pin %u can't be set into Peripheral Manager.", pin);
+    end();
 }
 
 void HWCDC::end()
 {
-    //Disable tx/rx interrupt.
+    //Disable/clear/free tx/rx interrupt.
     usb_serial_jtag_ll_disable_intr_mask(USB_SERIAL_JTAG_LL_INTR_MASK);
+    usb_serial_jtag_ll_clr_intsts_mask(USB_SERIAL_JTAG_LL_INTR_MASK);
```

Comment:
Won't clearing the interrupts here cause the same issue as #9316 if the HWCDC::begin() is called again?

Reply:
When the Serial CDC is ended, all data (in/out) must be flushed and the communication is terminated.
The line goes to "dead state". This is the Arduino standard.

Reply:
If the sketch calls `Serial.end()` followed by 2 seconds + `Serial.begin()`, but tries to send some bytes to the ESP32-S3/C3 using the Serial Monitor along those 2 seconds, the Serial Terminal will send it, HW Serial will populate its CDC Buffer (64 bytes) and it will wait until `Serial.begin()` starts the ISR. No data is lost along those 2 seconds.

Reply:
Yep sorry my bad that makes sense!